File: src/2917.find-the-k-or-of-an-array.3183/solution.ts
function findKOr (nums: number[], k: number): number {
    const counts = new Array<number>(32).fill(0);
    for (const num of nums) {
        const str = num.toString(2);
        for (let i = str.length - 1; i > -1; i--) {
            if (str[i] === '1') {
                counts[32 - (str.length - i)]++;
            }
        }
    }
    let result = 0;
    let base = 1;
    for (let i = 31; i > -1; i--) {
        if (counts[i] >= k) {
            result += base;
        }
        base *= 2;
    }
    return result;
}

File: src/2917.find-the-k-or-of-an-array.3183/solution.java
class Solution {
    public int findKOr(int[] nums, int k) {
        int result = 0;
        int offset = 1;
        for (int i = 0; i < 31; i++) {
            int cnt = 0;
            for (int n : nums) {
                if ((n & offset) != 0) {
                    cnt++;
                    if (cnt >= k) {
                        break;
                    }
                }
            }
            if (cnt >= k) {
                result |= offset;
            }
            offset <<= 1;
        }
        return result;
    }
}